broker-cli/cmd/bindings.go

```diff
@@ -59,6 +59,11 @@ var (
 			if err != nil {
 				log.Fatalf("Error creating binding %s to instance %s: %v", bindingsFlags.bindingID, bindingsFlags.instanceID, err)
 			}
+
+			if !bindingsFlags.acceptsIncomplete {
+				log.Fatal("Cannot create binding synchronously, set flag --asynchronous to be true")
```

Comment:
Just want to confirm, will "--asynchronous=true" work? Or "true" needs to be "True"?
And is there an alias for this flag so that we can say "set flag --asynchronous (-x) to be true"? (Not necessarily be "-x".)

Reply:
what if, instead of requiring the --asynchronous flag, the broker-cli simply retried in asynchronous mode?

Reply:
"--asynchronous true" does work. This has been tested. The short form is -a true

There are multiple things we can do here:
1) Assume all calls are asynchronous, i.e only support async create/delete,/update instance and create/ update binding. The only reason this "async" flag was added was to keep it compliant with OSB. There maybe brokers out there which support sync provisioning, or we may support sync provisioning in the future, so there is value in supporting sync mode
2) Make the flag "synchronous" and have the default value as false OR change the default value of the "asynchronous" flag to be true
3) what if, instead of requiring the --asynchronous flag, the broker-cli simply retried in asynchronous mode? I guess this an option too but the caller maybe be expecting an instance but may end up getting an operation. Not sure we need this right now.


I am leaning towards changing the default value of the "asynchronous" flag to be true

Reply:
Please respond with your thoughts.

Reply:
Maybe we should make async the default behavior and have a --sync flag instead? (Discussed offline with Deepika already)

Reply:
I am not sure how people like the idea of setting the default value of a flag to be true. Usually to use a boolean flag, we'll do `command --flag`. And if we don't want to use that flag/set it to be false, it is just `command`. Using `--flag false` seems to be not super user friendly.

IMO, instead of having the "--async" flag, we can use the "--sync" flag. So async calls become the default option.

Reply:
alright, you have convinced me. I am making the change.

Reply:
Talked offline. We can always add a --sync flag when needed.